subsys/bluetooth/services/ots/ots.c

```diff
@@ -351,6 +392,13 @@ int bt_ots_obj_delete(struct bt_ots *ots, uint64_t id)
 		return -EBUSY;
 	}
 
+	if (ots->cb->obj_deleted) {
+		err = ots->cb->obj_deleted(ots, NULL, obj->id);
```

Comment:
According to the spec, "The Object Client shall ensure that the object that it intends to delete from the Object Server has been selected as the Current Object."
That is the responsibility of the client, but the server does not seem to enforce it.

Reply:
The object creation and deletion paths local to the server do not follow all OTS specification requirement. For example the local object creation path doesn't leave the object's current size at 0 or the object's name as an empty string.

The object deletion path through the Bluetooth connection does ensure the current object is selected when it is being deleted. It would be fairly clunky for a server to have to select the object to delete it locally.